### ISSUE TITLE
Question about training loss

### ISSUE THREAD

**Issue by RuixiangZhao** [NVlabs/RADIO#122]
Could you provide more details about the training loss?

Below is a training process I have imagined based on the content in the paper (the example uses only one teacher):

```python
# Feature extraction
stu_cls_feature, stu_patch_features =  student_model(input)  # stu_cls_feature: [b, c1], stu_patch_features: [b, h1, w1, c1]
tea_cls_feature, tea_patch_features =  teacher_model(input)  # tea_cls_feature: [b, c2], tea_patch_features: [b, h2, w2, c2]

# Bilinear interpolation
stu_patch_features_new = bilinear_interpolation(stu_patch_features)  # stu_patch_features_new: [b, h2, w2, c1]

# Adapter head
stu_cls_feature_adapted = adaptor1(stu_cls_feature)  # stu_cls_feature_adapted: [b, c2]
stu_patch_features_adapted = adaptor2(stu_patch_features_new)  # stu_patch_features_adapted: [b, h2, w2, c2]

# Loss
loss_summary = cos_loss(stu_cls_feature_adapted, tea_cls_feature)
loss_spatial = 0.9 * cos_loss(stu_patch_features_adapted.reshape(-1, c2), tea_patch_features.reshape(-1, c2)) + 0.1 * smooth_l1(stu_patch_features_adapted, tea_patch_features)
total_loss = loss_summary + lamda * loss_spatial
```

For the above process, do you see any misunderstandings, especially regarding the `loss_spatial` part? I would appreciate any feedback you might have!

**Comment by RuixiangZhao** [comment]
Thank you for your prompt response, which have cleared up my confusion.
